src/scipp/reduction.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+import uuid
+from typing import Union, List, Tuple
+from .typing import VariableLike
+from .core import concat, reduction
+
+
+def _make_method(f):
+    def method(self) -> VariableLike:
+        return f(self._obj, self._dim)
+
+    return method
+
+
+class BinsReducer:
+    def __init__(self, obj: VariableLike, dim: str):
+        self._obj = obj
+        self._dim = dim
+
+    def concat(self):
+        return self._obj.bins.concat(self._dim)
+
+
+class Reducer:
+    all = _make_method(reduction.all)
+    any = _make_method(reduction.any)
+    max = _make_method(reduction.max)
+    min = _make_method(reduction.min)
+    sum = _make_method(reduction.sum)
```

Comment:
How much time does this save compared to `a + b + c`? I am not sure it is enough to justify the extra memory usage.

Reply:
- It provides a consistent interface. Users may of course choose to operate differently in a specific case.
- Given this interface, we may later switch to a more efficient implementation, such as what you suggest, or concatenating in pieces to avoid large memory usage.